Netable/Netable/Netable.swift

```diff
@@ -91,21 +97,23 @@ open class Netable {
             if T.Parameters.self != Empty.self {
                 try urlRequest.encodeParameters(for: request, defaultEncodingStrategy: config.jsonEncodingStrategy)
             }
+
+            config.globalHeaders.forEach { key, value in
+                urlRequest.setValue(value, forHTTPHeaderField: key)
+            }
+
+            request.headers.forEach { key, value in
+                urlRequest.setValue(value, forHTTPHeaderField: key)
+            }
+
+            return try await startRequestTask(request, urlRequest: urlRequest, id: UUID().uuidString)
         } catch {
             let netableError = (error as? NetableError) ?? NetableError.unknownError(error)
             await log(.requestCreationFailed(urlString: request.path, error: netableError))
+            self.requestFailureDelegate?.requestDidFail(request, error: netableError)
```

Comment:
It's probably going to be a bit of a pain, but I think like the log, this delegate should probably be switched to being an async protocol, as it is right now, I think this call can happen on any thread/actor, which people might not be expecting if the protocol isn't async. I suspect that's also going to be an error once we do get the concurrency checking enabled.